pkg/reconciler/instances/connectivityproxy/command.go

```diff
@@ -134,5 +135,18 @@ func (a *CommandActions) Remove(context *service.ActionContext) error {
 	if err != nil {
 		return errors.Wrap(err, "Error during removal")
 	}
+
+	context.Logger.Info("Deleting secrets")
+
+	_, err = context.KubeClient.DeleteResource(context.Context, "secret", "cc-certs", "istio-namespace")
```

Comment:
I performed a test and noticed that secrets are not deleted. I think the reason is that service account for Connectivity Proxy doesn't have delete permission. Please see [this](https://github.com/kyma-project/control-plane/blob/main/resources/kcp/charts/component-reconcilers/templates/rbac.yaml).